subscriptions-manager/src/main/java/io/arlas/subscriptions/dao/ElasticUserSubscriptionDAOImpl.java

```diff
@@ -140,4 +158,68 @@ public void initSubscriptionIndex() throws ArlasSubscriptionsException {
             throw new ArlasSubscriptionsException(arlasSubscriptionIndex  + " elasticsearch index does not exist, create it to run ARLAS-Subscription");
         }
     }
+
+    public void initBulkProcessor(final long total) throws ArlasSubscriptionsException {
+        this.initSubscriptionIndex();
+
+        bulkProcessor = BulkProcessor.builder(
+                client,
+                new BulkProcessor.Listener() {
+                    long count = 0;
+
+                    @Override
+                    public void beforeBulk(long executionId,
+                                           BulkRequest request) {
+                        count += request.numberOfActions();
+                    }
+
+                    @Override
+                    public void afterBulk(long executionId,
+                                          BulkRequest request,
+                                          BulkResponse response) {
+                        logger.info("Nb of doc synchronised: " + count + "/" + total + " (" + df2.format((double) count/total) + ")");
+                        if (response.hasFailures()) {
+                            logger.warn("-> But some failure in synchronization: " + response.buildFailureMessage());
+                        }
+                    }
+
+                    @Override
+                    public void afterBulk(long executionId,
+                                          BulkRequest request,
+                                          Throwable failure) {
+                        logger.info("Nb of doc synchronised: " + count + "/" + total + " (" + df2.format((double) count/total) + ")");
+                        logger.warn("-> Failure in synchronization: " + failure.getMessage());
+                    }
+                })
+                .setConcurrentRequests(0)
+                .setBulkActions(10000)
+                .setBulkSize(new ByteSizeValue(5, ByteSizeUnit.MB))
```

Comment:
If we're not sure of the best config and this is to be tested, this could be put in the configuration file.

Reply:
Done